### DISCUSSION TITLE
프로젝트 회고록

### DISCUSSION THREAD

**Discussion by zangsu** [zangsu/cs-test-service#34] [Announcements]
프로젝트가 모두 완성되었습니다.
소감 및 성찰 관련 회고록을 작성해 주세요!

**Comment by Moon-GD** [comment]
프로젝트 전에 세웠던 목표 달성률 점검도 필요할 것 같아요.
최종 목표 달성률은 <strong>83.33%</strong> (5/6) 입니다.

## FE
- ~~<i>코드 formatter 활용 (eslint & prettier)</i>~~
- ~~<i>React 숙달</i>~~
- redux 활용

## BE
- ~~<i>서버호스팅 (AWS)</i>~~
- ~~<i>스프링 개발 (MVC 패턴 연습)</i>~~
- ~~<i>서버 통신</i>~~

FE에서 redux를 활용하지 못했는데 <a href="https://github.com/Start-as-Web-Developers/tumblbug-clone-coding">다음 프로젝트</a>에서 꼭 활용하고자 합니다.

**Comment by zangsu** [comment]
# 프로젝트 기간 준수 실패
- 도메인을 연결하여 HTTPS 프로토콜 통신을 하는 과정에서 502 Bad Gateway 발생
- 로그 분석 및 문제 해결 과정이 예상보다 오래 걸리면서 기간 준수 실패
- [트러블 슈팅 과정](https://velog.io/@zangsu/AWS-502-Bad-Gateway-1)

<br>

## 문제 원인
- HTTPS 통신을 위해 AWS ALB 사용
- ALB의 healthy check에서 서버를 unhealthy로 인식
  - ALB healthy check는 `GET "도메인"/`를 사용
  - `GET "도메인"/` 요청을 처리하는 컨트롤러 부재
  - 해당 요청에서 404 에러 발생
- 요청 정보를 서버로 전달하지 않고, 502 Bad Gateway 응답

<br>

## 문제 해결
- 문제 해결 방법
  - healthy check 경로 재설정
  - healthy check 요청을 처리할 수 있는 컨트롤러 구현

기존 구현 컨트롤러는 모두 특정 기능을 처리하기 위한 URL이었기 때문에, `GET "도메인"/` 요청을 처리하는 컨트롤러 구현

**Comment by zangsu** [comment]
# 개선할 점
- Branch 구분이 부실했음
  - 해당 branch와 관련 없는 코드 수정
- 학습 내용 공유가 이루어지지 않음
